pyomo/repn/plugins/gams_writer.py

```diff
@@ -326,15 +333,17 @@ def _write_model(self,
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
-                    if v.bounds == (0,1):
+                    if (v.has_lb() and (value(v.lb) >= 0)) and \
+                       (v.has_ub() and (value(v.ub) <= 1)):
                         binary.append(var)
                     else:
                         ints.append(var)
-                elif v.lb == 0:
+                elif value(v.lb) == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
             else:
+                # GH: doesn't seem like this block is ever hit
```

Comment:
(Pretty sure) that would be because none of the solver tests use expressions in their constraints. I've used this code block in my own local tests in the past.